modules/puppet_proxy/configuration_loader.rb

```diff
@@ -0,0 +1,20 @@
+module ::Proxy::Puppet
+  class ConfigurationLoader
+    def load_programmable_settings(settings)
+      raise ::Proxy::Error::ConfigurationError, "Parameter ':puppet_version' is expected to have a non-empty value" if settings[:puppet_version].to_s.empty?
```

Comment:
Can't rely on that here -- validators are run after settings are loaded and processed, and I need :puppet_version setting value here. It's either run validators several times at different stages of plugin initialization process (there are issues with this though) or do on the spot checks like this one.